pike/test/querydirectory.py

```diff
@@ -124,3 +125,277 @@ def test_restart_scan(self):
+        # 66 bytes is the struct size of FILE_DIRECTORY_INFORMATION
+        self.n_entries = (self.payload_size /
+                     ((len(self.filename_pattern) - 1) * 2 + 66)) + 1
+        self.info("Creating {0} files to fill {1} bytes".format(
+                        self.n_entries,
+                        self.payload_size))
+
+        chan, tree = self.tree_connect()
+        with self.get_test_root(chan, tree) as root_handle:
+            pass            # ensure the test root dir is created
+        # Creates files within the root directory.
+        remaining_files = self.n_entries
+        while remaining_files > 0:
+            batch_futures = []
+            batch_size = 30
+            if remaining_files < batch_size:
+                batch_size = remaining_files
+            for ix in xrange(batch_size):
+                filename = self.filename_pattern.format(remaining_files)
+                path = "{0}\\{1}".format(self.root_dir_name, filename)
+                batch_futures.append((filename, chan.create(tree, path)))
+                remaining_files -= 1
+            for filename, fu in batch_futures:
+                fh = fu.result()
+                chan.close(fh)
+                self.filenames.append(filename)
+        chan.logoff()
+        QueryDirectoryTestMaxMtu.dataset_created = True
+
+    def setUp(self):
+        self.create_dataset()
+
+    @contextmanager
+    def get_test_root(self, chan, tree):
+        root_handle = chan.create(
+            tree,
+            self.root_dir_name,
+            access=pike.smb2.GENERIC_READ,
+            options=pike.smb2.FILE_DIRECTORY_FILE,
+            share=pike.smb2.FILE_SHARE_READ |\
+                  pike.smb2.FILE_SHARE_WRITE |\
+                  pike.smb2.FILE_SHARE_DELETE).result()
+        try:
+            yield root_handle
+        finally:
+            chan.close(root_handle)
+
+    # Enumerate directory at FILE_DIRECTORY_INFORMATION level using the maximum transfer size.
+    def gen_file_directory_info_max_mtu(self, dialect=None, caps=0):
+        """
+        Enumerate directory at FILE_DIRECTORY_INFORMATION level using the maximum transfer size.
+        """
+
+        with self.tree_connect_with_dialect_and_caps(dialect, caps) as (chan, tree):
+            # Retrieves the maximum transaction size to determine how big the command's payload can be (MTU).
+            max_trans_size = chan.connection.negotiate_response.max_transact_size
+
+            # Enumerates the root directory and validates that the expected files are present in it.
+            with self.get_test_root(chan, tree) as root_handle:
+                dir_query = chan.query_directory(
+                    root_handle,
+                    file_information_class=pike.smb2.FILE_DIRECTORY_INFORMATION,
+                    flags=0,
+                    file_index=0,
+                    file_name='{0}*'.format(self.filename_prefix),
+                    output_buffer_length=self.payload_size)
```

Comment:
Shouldn't we be using max_trans_size here rather than payload_size?  (Isn't the goal here to test that we can do a full max_trans_size transfer?)

Reply:
well windows 10 / windows server 2016 return 8mb for max_trans_size, but will still fail the request if the client asks for a buffer larger than 64k, so i'm not really sure why those servers are returning such a large max_trans_size. any ideas would be welcome